### ISSUE TITLE
aws_opsworks_application.short_name attempts update in-place and fails silently

### ISSUE THREAD

**Issue by omnibs** [terraform-providers/terraform-provider-aws#6306]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.11.7
+ provider.aws v1.41.0
```

### Affected Resource(s)

* aws_opsworks_application

### Terraform Configuration Files

```hcl
resource "aws_opsworks_application" "foo-app" {
  name        = "foobar application"
  short_name  = "foobar"
  stack_id    = "${aws_opsworks_stack.main.id}"
  type        = "rails"
  description = "This is a Rails application"

  domains = [
    "example.com",
    "sub.example.com",
  ]

  environment = {
    key    = "key"
    value  = "value"
    secure = false
  }

  app_source = {
    type     = "git"
    revision = "master"
    url      = "https://github.com/example.git"
  }

  enable_ssl = true

  ssl_configuration = {
    private_key = "${file("./foobar.key")}"
    certificate = "${file("./foobar.crt")}"
  }

  document_root         = "public"
  auto_bundle_on_deploy = true
  rails_env             = "staging"
}
```

### Debug Output

Sorry, I think this is short enough to just paste:
```
  1 Shortname was inconsistent with current Stacks
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ aws_opsworks_application.foo-app
      short_name: "foobar" => "FooBar"


Plan: 0 to add, 1 to change, 0 to destroy.
```

### Expected Behavior

Either shortname should have changed or resource should have been recreated.

### Actual Behavior

Terraform said it'd change the shortname but it stayed the same.

I checked in AWS Console and apparently you can't change shortname.

### Steps to Reproduce

1. `terraform apply` with some shortname
2. Change the shortname to something else
3. `terraform apply`
4. Check that the app remains with the same shortname

### Important Factoids

None

### References

None

**Comment by bflad** [comment]
The fix for properly triggering resource recreation on `short_name` updates has been merged and will release with version 1.43.0 of the AWS provider, likely later today or tomorrow. 👍

**Comment by omnibs** [comment]
Rad! Thanks for the quick fix!

**Comment by bflad** [comment]
This has been released in [version 1.43.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1430-november-07-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
